js/app/service/eventseditordialogservice.js

```diff
@@ -24,145 +24,174 @@
+			if (template === EDITOR_SIDEBAR) {
+				angular.element('#app-content').removeClass('with-app-sidebar');
+			}
 
-				rejectDialog(reason);
-			});
-		}
+			if (reason !== 'superseded') {
+				context.cleanup();
+			}
 
-		function openRepeatQuestion() {
-			//TODO in followup PR
-		}
+			unlock();
+			reject(reason);
+		});
+	};
 
-		this.open = function(scope, fcEvent, positionCallback, lockCallback, unlockCallback) {
-			const skipPopover = (angular.element('#fullcalendar').attr('data-skipPopover') === 'yes');
+	context.openRepeatQuestion = () => {
+		// TODO in followup PR
+	};
 
-			//don't reload editor for the same event
-			if (self.fcEvent === fcEvent) {
-				return self.promise;
-			}
+	/**
+	 * open dialog for editing events
+	 * @param {object} scope
+	 * @param {FcEvent} fcEvent
+	 * @param {positionCallback} calculatePosition
+	 * @param {lockCallback} lock
+	 * @param {unlockCallback} unlock
+	 * @returns {Promise}
+	 */
+	this.open = function(scope, fcEvent, calculatePosition, lock, unlock) {
+		const skipPopover = (angular.element('#fullcalendar').attr('data-skipPopover') === 'yes');
+
+		// don't reload editor for the same event
+		if (context.fcEvent === fcEvent) {
+			return context.promise;
+		}
 
-			//is an editor already open?
-			if (self.fcEvent) {
-				self.eventModal.dismiss('superseded');
-			}
+		// dismiss existing dialogs
+		if (context.fcEvent) {
+			context.eventModal.dismiss('superseded');
+		}
 
-			cleanup();
-			self.promise = new Promise(function(resolve, reject) {
-				self.fcEvent = fcEvent;
-				self.simpleEvent = fcEvent.getSimpleEvent();
-				if (fcEvent.vevent) {
-					self.calendar = fcEvent.vevent.calendar;
-				}
-				self.scope = scope;
+		context.promise = new Promise(function(resolve, reject) {
+			// calculate position of popover
+			// needs to happen before locking event
+			const position = calculatePosition();
 
-				//calculate position of popover
-				var position = positionCallback();
+			// lock new fcEvent
+			lock();
 
-				//lock new fcEvent and unlock old fcEvent
-				lockCallback();
-				if (self.unlockCallback) {
-					self.unlockCallback();
-				}
-				self.unlockCallback = unlockCallback;
+			const calendar = (fcEvent.vevent) ? fcEvent.vevent.calendar : null;
+			const simpleEvent = fcEvent.getSimpleEvent();
 
-				//skip popover on small devices
-				if (angular.element(window).width() > 768 && !skipPopover) {
-					openDialog(EDITOR_POPOVER, position, reject, resolve);
-				} else {
-					openDialog(EDITOR_SIDEBAR, null, reject, resolve);
-				}
-			});
+			// skip popover on small devices
+			if (context.showPopover() && !skipPopover) {
+				context.openDialog(EDITOR_POPOVER, resolve, reject, unlock, position, scope, fcEvent, simpleEvent, calendar);
+			} else {
+				context.openDialog(EDITOR_SIDEBAR, resolve, reject, unlock, [], scope, fcEvent, simpleEvent, calendar);
+			}
+		});
 
-			return self.promise;
+		return context.promise;
 	};
 
-	return this;
+	/**
```

Comment:
So my IDE knows what parameters the callbacks [up there](https://github.com/nextcloud/calendar/pull/179/files#diff-a6601ae4e3cfb5427aee447d40ee87ccR140) expect